modules/persistence/src/services/metadata/repos_branches/index.ts

```diff
@@ -0,0 +1,98 @@
+import { pool, db } from '../../connector';
+import { Metadata } from '..';
+import { project } from '../ToC';
+import { WithId } from 'mongodb';
+
+type EnvKeyedObject = {
+  prd: any;
+  preprd: any;
+  dotcomstg: any;
+  dotcomprd: any;
+};
+
+export interface BranchEntry {
+  name: string;
+  gitBranchName: string;
+  [key: string]: any;
+}
+
+export interface ReposBranchesDocument extends WithId<Document> {
+  repoName: string;
+  project: string;
+  branches: BranchEntry[];
+  url: EnvKeyedObject;
+  prefix: EnvKeyedObject;
+  [key: string]: any;
+}
+
+const internals: { [key: project]: ReposBranchesDocument } = {};
+
+// Queries pool*.repos_branches for all entries for associated_products in a shared metadata entry
+export const getAllAssociatedRepoBranchesEntries = async (metadata: Metadata) => {
+  const { associated_products } = metadata;
+  if (!associated_products || !associated_products.length) return [];
```

Comment:
nit: we can do something like `const { associated_products = [] }` since the destructure is for only one property here

Reply:
good catch! i also was not using the lists of `res` and `fetch` as intended, and fixed in the latest commit.